src/apps/natdiscovery/natdiscovery.c

```diff
@@ -86,7 +86,7 @@ static int stunclient_send(int sockfd, ioa_addr *local_addr, int *local_port, io
 
   if (response_port >= 0) {
```

Comment:
Another condition that becomes always true

Reply:
To address this you'll need to.

1. Find all places where this function is called, and determine how they could potentially get an invalid port number to pass into this function.
2. It's possible that some callers are *intentionally* passing `-1` into the function to indicate an invalid port. In that case, you'll want to change the parameter to a pointer-to-uint16_t, and use `NULL` to indicate "not passed", instead of `-1`.
3. If none of the callers are intentionally passing invalid ports, but are only able to pass invalid ports because they read an invalid port out of a config file / network packet or something, then you need to put the error handling at the place where the bad valid is acquired.
4. Assuming that none of the callers are intentionally passing invalid ports, then remove the if-check, and make the body of the conditional always happen.